crates/subspace-gateway/src/commands/http/server.rs

```diff
@@ -0,0 +1,151 @@
+use actix_web::{web, App, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Deserializer, Serialize};
+use std::default::Default;
+use std::sync::Arc;
+use subspace_core_primitives::hashes::{blake3_hash, Blake3Hash};
+use subspace_core_primitives::pieces::PieceIndex;
+use subspace_core_primitives::BlockNumber;
+use subspace_data_retrieval::object_fetcher::ObjectFetcher;
+use subspace_data_retrieval::piece_getter::PieceGetter;
+use tracing::{debug, error, trace};
+
+pub(crate) struct ServerParameters<PG>
+where
+    PG: PieceGetter + Send + Sync + 'static,
+{
+    pub(crate) object_fetcher: ObjectFetcher<PG>,
+    pub(crate) indexer_endpoint: String,
+    pub(crate) http_endpoint: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+struct ObjectMapping {
+    hash: Blake3Hash,
+    piece_index: PieceIndex,
+    piece_offset: u32,
+    #[serde(deserialize_with = "string_to_u32")]
+    block_number: BlockNumber,
+}
+
+fn string_to_u32<'de, D>(deserializer: D) -> Result<u32, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    s.parse::<u32>().map_err(serde::de::Error::custom)
+}
+
+async fn request_object_mappings(endpoint: String, key: String) -> anyhow::Result<ObjectMapping> {
+    let client = reqwest::Client::new();
+    let object_mappings_url = format!("http://{}/objects/{}", endpoint, key,);
```

Comment:
Hardcoding the HTTP protocol would avoid the usage of indexers with HTTPS endpoints? For example, using the current`https://indexer.taurus.autonomys.xyz`.

Reply:
Using this indexer could be useful for debugging and testing purposes

Reply:
I agree with this suggestion. I'll create and test a separate PR.